core/async/src/time.rs

```diff
@@ -150,17 +174,19 @@ impl FakeClockInner {
         if d == Duration::ZERO {
             return;
         }
-        let now = *self.mono.borrow();
-        self.mono.send(now + d).unwrap();
+        self.instant += d;
         self.utc += d;
+        while let Some(earliest_waiter) = self.waiters.peek() {
+            if earliest_waiter.deadline <= self.instant {
+                self.waiters.pop().unwrap().waker.send(()).ok();
+            } else {
+                break;
+            }
+        }
```

Comment:
super nit: you can use `peek_mut` and then call [pop](https://doc.rust-lang.org/std/collections/binary_heap/struct.PeekMut.html#method.pop) on it in order to avoid `unwrap `later, which is a bit cleaner